sdk/eventhub/Azure.Messaging.EventHubs/src/EventProcessorClient.cs

```diff
@@ -287,194 +437,404 @@ public class EventProcessorClient : EventProcessorBase<PartitionContext>, IAsync
+
+                            ActiveLoadBalancingTask = RunAsync(RunningTaskTokenSource.Token);
+                        }
+                    }
                 }
-                else if (ownership.SequenceNumber.HasValue)
+                finally
                 {
-                    startingPosition = EventPosition.FromSequenceNumber(ownership.SequenceNumber.Value);
+                    RunningTaskSemaphore.Release();
                 }
-
-                // TODO: it might be troublesome to let the users add running tasks by themselves.  If the user adds a custom
-                // processing task that's not RunPartitionProcessingAsync, how would the base stop it?  It would not have a cancellation
-                // token to do so.
-
-                ActivePartitionProcessors[context.PartitionId] = RunPartitionProcessingAsync(context.PartitionId, startingPosition, Options.MaximumReceiveWaitTime, Options.RetryOptions, Options.TrackLastEnqueuedEventProperties);
-            }
-            catch (Exception)
-            {
-                // If processing task creation fails, we'll try again on the next time this method is called.  This should happen
-                // on the next load balancing loop as long as this instance still owns the partition.
-                // TODO: delegate the exception handling to an Exception Callback.  Do we really need a try-catch here?
             }
         }
 
         /// <summary>
-        ///   The handler to be called once event processing stops for a given partition.
+        ///   Starts the event processor.  In case it's already running, nothing happens.
+        /// </summary>
+        ///
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> instance to signal the request to cancel the operation.</param>
+        ///
+        /// <exception cref="EventHubsClientClosedException">Occurs when this <see cref="EventProcessorClient" /> instance is already closed.</exception>
+        /// <exception cref="InvalidOperationException">Occurs when this method is invoked without <see cref="ProcessEventAsync" /> or <see cref="ProcessErrorAsync" /> set.</exception>
+        ///
+        public virtual void StartProcessing(CancellationToken cancellationToken = default) => StartProcessingAsync(cancellationToken).GetAwaiter().GetResult();
+
+        /// <summary>
+        ///   Stops the event processor.  In case it isn't running, nothing happens.
         /// </summary>
         ///
-        /// <param name="reason">The reason why the processing for the associated partition is being stopped.</param>
-        /// <param name="context">The context in which the associated partition was being processed.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> instance to signal the request to cancel the operation.</param>
         ///
         /// <returns>A task to be resolved on when the operation has completed.</returns>
```

Comment:
We've typically avoided these `<return>` for just `Task`, since they offer no real value in understanding the context or intended usage of a return value;  In the case where we have a `Task<T>` we normally describe the `T` that gets returned and assume the task is an implicit thing.

Reply:
Removed it from every `EventProcessorClient`-related method, except for this one:

https://github.com/Azure/azure-sdk-for-net/blob/68b1a07746e77556f7c5bf64f5f387acce37b747/sdk/eventhub/Azure.Messaging.EventHubs/src/EventProcessorClient.cs#L1157-L1159

Because I think it contains a useful description.

Reply:
Agree; that is helpful in this context.